test/unit/sync_checker/checks/details_check_test.rb

```diff
@@ -0,0 +1,145 @@
+require 'minitest/autorun'
+require 'mocha/setup'
+require 'active_support'
+require 'active_support/json'
+require 'active_support/core_ext'
+
+require_relative '../../../../lib/sync_checker/checks/details_check'
+
+module SyncChecker::Checks
+  class DetailsCheckTest < Minitest::Test
+    def test_non_200_returns_an_empty_array
+      response = stub(
+        response_code: 404,
+        body: {
+        }.to_json
+      )
+
+      expected = {
+        body: "a bit of html"
+      }
+
+      assert_equal [], DetailsCheck.new(expected).call(response)
```

Comment:
That's a good point. We've tried to make it so as an individual 'check' only returns errors if the thing it is checking is actually 'wrong' and returns none otherwise. We did have a specific `status` check at one point. We probably need to reinstate that.